### ISSUE TITLE
[Bug]: Web content flashes when Firefox opened

### ISSUE THREAD

**Issue by FormularSumo** [mozilla-mobile/fenix#22602]
### Steps to reproduce

1. Set system navigation to gesture navigation
2. Have a web page open
3. Switch to another application
4. Wait a few seconds or interact with the app to change focus
5. Switch back to Firefox

### Expected behaviour

Firefox should open and resume without web content flashing

### Actual behaviour

Web content flashes to black and then back to content upon reopening Firefox

https://user-images.githubusercontent.com/36278229/143559360-f7dbca96-7718-44ea-a881-94d0da37e257.mp4

### Device name

Google Pixel 4a

### Android version

Android 12

### Firefox release type

Firefox Nightly

### Firefox version

96.0a1

### Additional information

The bug also happens in Firefox for Android Beta and Firefox for Android. I used fresh installations for testing those, no extensions and no settings changed. As long as the focus has changed to a different app before switching back, this happens every time I switch back to Firefox. It doesn't happen when opening Firefox from homescreen or when using 3 button navigation. The content flashing seems to be tied to the navigation bar flashing. The bug doesn't happen in Chrome depsite the navigation bar also flashing.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-18754)

**Comment by SoftVision-LorandJanos** [comment]
Hi @FormularSumo! Thank you for your report!
I could reproduce the issue you described.
Tested on the:
- latest Nightly (2021-11-29).
- latest Beta (95.0.0-beta6).
- latest Release (94.1.2).

Device used: Xiaomi Mi11 Lite (Android11).

**Comment by FormularSumo** [comment]
https://user-images.githubusercontent.com/36278229/163852422-1e880077-dd3e-4fef-9049-88ffd9d97a2b.mp4

I noticed that I no longer get this issue, tested on latest nightly 101, beta 100 and release 99 (fresh installs).
I also downloaded Firefox 94.1.2 from when I reported the issue and I don't get the problem there anymore.
The only reason I could think of for this is that now my phone is on Android 12L, so I tried running Firefox release 99 on an Android 12 and Android 12L Pixel 5 virtual machine in Android Studio but found that the issue still happened in Android 12L, as well as 12. However on my brother's phone, a Pixel 4a running Android 12L like mine, the issue doesn't happen.
